### ISSUE TITLE
Torrent

### ISSUE THREAD

**Issue by c-sanchez** [erickutcher/httpdownloader#21]
Hi erickutcher, I wonder if possible in the future add support for Torrent, I guess this is a bit complicated to do, but indeed there are some simple download managers wich have this, Like EagleGet and MiniGet, this last the latter is very small too.

**Comment by smaragdus** [comment]
No one is asking me but for me this is not a good idea- there are multiple torrent clients with specific features which will do the job better. For me such an option is irrelevant- as far as I know this program is intended to be a tiny, light and fast download manager without unnecessary bloat. I suppose that torrent support will be an example of [feature creep](https://en.wikipedia.org/wiki/Feature_creep) but the developer knows better. To this day I still use Free Download Manager but I have not downloaded a single torrent with it because I prefer a download manager to be just a download manager, not a torrent client.

**Comment by c-sanchez** [comment]
Well,Torrent support for download manager I think don't makes this bloated, I guess just increase the size/options of program of course.
Free Download Manager it's a good example about how make bloated an software, while the old 3.x versions are Windows only, this is lightweight, but the new 5.x version its bigger, making this an cross-platform Download Manager in QT, but both 5.x and 3.0 versions have support for torrents too, although I'm agree this is not the best tool to download this :P
But as feature Torrent I think is pretty nice, by example if someone wan't make a queue beetwen torrents and direct link downloads.
By that I've comment MiniGet as example, it's an download manager with torrent support in 2mb~ heh
But in any case I understand your point :)

**Comment by smaragdus** [comment]
@c-sanchez 

Free Download Manager version 3 was close to perfect for me- feature-rich, extremely customizable and light. Unfortunately the developer(s) chose to severely dumb it down so they ruined it completely. I am still using version 3 because it integrates perfectly with my old pre-Quantum version of Firefox.

Since I was searching for a replacement of Free Download Manager I once tested MiniGet but did not like it mainly because the so called portable version is not portable writing a lot of data to Windows registry.

I like HTTP Downloader for what it does- downloading files via HTTP/HTTPS. I do not need support for other protocols- BitTorrent, eMule, FTP, MMS, RTMP, RTSP, etc., for such needs I use other tools. This is just my opinion.

**Comment by c-sanchez** [comment]
Yeah @smaragdus I'm agree with you about FDM development.
Heh I had not noticed what you say about MiniGet, good point.
But yeah, Indeed for now I'm using IDM to download, I had tried replace it with EagleGet which imho is really good, but unafortunatelly the last versions I still find them somewhat unstable and problematic, so..

I think with some of luck I will use some new version of HTTP Downloader for my daily use :)
And torrents.. something more haha, that is not a big deal

Regards.

**Comment by erickutcher** [comment]
I'd have to change the name of the program. :-)

Adding torrent support would be a monumental task that I just can't devote time to. It would more than likely involve an entire rewrite. And at that point, it would make more sense just to build a dedicated client. It's interesting to think about, but I'll have to pass.

**Comment by smaragdus** [comment]
@c-sanchez 

Depending on your needs and tastes you may choose between these torrent clients:

- [BiglyBT](https://www.biglybt.com/) (open source, stable, no portable option/version, faster development, Java);
- [Deluge](https://deluge-torrent.org/) (open source, stable, portable option, slower development);
- [Halite](http://www.binarynotions.com/halite-bittorrent-client/) (open source, stable, portable version, slower development);
- [PicoTorrent](https://picotorrent.org/) (open source, unstable, portable version, slower development);
- [qBittorrent](https://www.qbittorrent.org/) (open source, stable, portable option, faster development);
- [Tixati](https://www.tixati.com/) (closed source, stable, portable version, faster development);
- [Transmission](https://transmissionbt.com/) (open source, stable, no portable option/version, slower development);

I do not recommend adware clients like [BitComet](https://www.bitcomet.com/en) and [μTorrent](https://www.utorrent.com/)

**Comment by smaragdus** [comment]
@erickutcher 

This is off-topic but I am curious- have you ever considered creating a simple GUI for **youtube-dl** ([Website](https://rg3.github.io/youtube-dl/), [GitHub](https://github.com/rg3/youtube-dl/), [Supported sites](https://rg3.github.io/youtube-dl/supportedsites.html))? In fact there are many but all of them are either poorly written or abandoned or both. Would this require too much work? Would it be possible to use some of the **HTTP Downloader** code?

**Comment by erickutcher** [comment]
I don't think it would be difficult, but it would be time consuming. Lots of GUI frontends are written in C# because it's quick to get going. Win32 C/C++ GUIs take 10 times longer to write. Some of my code could be repurposed.

I'm not a big video consumer, so I don't think I'd get much use out of it. And that's usually why I write programs.

**Comment by smaragdus** [comment]
@erickutcher 

I came across amother [youtube-dl](https://github.com/rg3/youtube-dl/) GUI- [YDL-UI](https://github.com/Maxstupo/ydl-ui)- it is written in C# but it works decently, since [2.0.0](https://github.com/Maxstupo/ydl-ui/releases/tag/v2.0.0) it has portable version too and it comes bundled with **youtube-dl** and **ffmpeg**.

Before I was using another one- [Youtube-DLG](https://github.com/MrS0m30n3/youtube-dl-gui)- it is written in Python, but the program is buggy, the development- slow and the developer discards issues which do not follow his [template](https://github.com/MrS0m30n3/youtube-dl-gui/issues/new) so I gave up.

**Comment by erickutcher** [comment]
College professors migrated from teaching C/C++ to Java because it was the popular thing at the time, and now they've moved onto Python for the same reason. So you end up seeing a lot of godawful GUI programs written in Java and Python because that's all anyone was taught. They're just not good languages for GUIs. /rant

**Comment by smaragdus** [comment]
Some time ago I came upon a decent [youtube-dl](https://github.com/ytdl-org/youtube-dl) GUI- **Vivid**l:

- [GitHub](https://github.com/Bluegrams/Vividl)
- [SourceForge](https://sourceforge.net/projects/vividl/)
- [Download](https://sourceforge.net/projects/vividl/files/)

**Screen**

![Vividl 0 1 0 0 - 2020-01-25 - 044](https://user-images.githubusercontent.com/9025757/75398913-95ba2600-5903-11ea-89a8-49d791c9dbb9.png)

It is C#/.NET program but it has portable version. It is in early stage of development but it works fine. For me it is the most visually pleasing GUI for [youtube-dl](https://github.com/ytdl-org/youtube-dl).

**Comment by erickutcher** [comment]
That looks like the Arc theme on Linux systems.

**Comment by PCUser2021** [comment]
Please close this post. There are already numerous BitTorrent clients and download managers that support BitTorrent, so adding this feature to HDL would only increase its size and introduce various problems and bugs. Currently, the best BitTorrent client available is qBittorrent. Please refrain from asking any further questions about this!
